src/ansys/units/dimensions.py

```diff
@@ -1,142 +1,92 @@
-        for idx, dim in enumerate(dimensions):
-            if dim == 1.0:
-                units += f" {unit_sys[idx]}"
-            elif dim != 0.0:
-                dim = int(dim) if dim % 1 == 0 else dim
-                units += f" {unit_sys[idx]}^{dim}"
-
-        return dimensions, condense(units=units)
-
-    def _units_to_dim(
-        self, units: str, power: float = None, dimensions: list = None
-    ) -> list:
-        """
-        Convert a unit string into a dimensions list.
-
-        Parameters
-        ----------
-        units : str
-            Unit string of quantity.
-        power : float
-            Power of unit string.
-
-        Returns
-        -------
-        list
-            Dimensions representation of unit string.
-        """
-        power = power or 1.0
-        dimensions = dimensions or [0.0] * self.max_dim_len()
-
-        # Split unit string into terms and parse data associated with individual terms
-        for term in units.split(" "):
-            _, unit_term, unit_term_power = filter_unit_term(term)
-
-            unit_term_power *= power
-
-            # retrieve data associated with fundamental unit
-            if unit_term in ansunits._fundamental_units:
-                idx = (
-                    ansunits._dimension_order[
-                        ansunits._fundamental_units[unit_term]["type"]
-                    ]
-                    - 1
-                )
-                dimensions[idx] += unit_term_power
-
-            # Retrieve derived unit composition unit string and factor.
-            if unit_term in ansunits._derived_units:
-                # Recursively parse composition unit string
-                dimensions = self._units_to_dim(
-                    units=ansunits._derived_units[unit_term]["composition"],
-                    power=unit_term_power,
-                    dimensions=dimensions,
-                )
-
-        return dimensions
-
-    @property
-    def units(self):
-        """Unit string representation of dimensions."""
-        return self._unit
+    def __init__(self, dimensions_container: dict[BaseDimensions : int | float] = {}):
+        self._dimensions = dimensions_container.copy()
+        for x, y in dimensions_container.items():
+            if not isinstance(x, BaseDimensions):
+                raise DimensionsError.INCORRECT_DIMENSIONS()
+            if y == 0:
+                del self._dimensions[x]
 
     @property
     def dimensions(self):
-        """Dimensions list."""
-        return self._dimensions
-
-    @staticmethod
-    def max_dim_len():
-        """Maximum number of elements within a dimensions list."""
-        return 9
+        """Dimensions as dictionary."""
+        dims = {x.value: y for x, y in self._dimensions.items()}
+        return dict(dims)
+
+    def __str__(self):
+        dims = {x.name: y for x, y in self._dimensions.items()}
+        if not dims:
+            dims = ""
+        return str(dims)
+
```

Comment:
You'll need to tweak these implementations - see my comment in unit.y. And you'll also want to add `__truediv__` and `__pow__`.